internal/pkg/aws/s3/s3.go

```diff
@@ -147,3 +148,16 @@ func (s *S3) EmptyBucket(bucket string) error {
 		listParams.VersionIdMarker = listResp.NextVersionIdMarker
 	}
 }
+
+// ParseURL parses S3 object URL and returns the bucket name and the key.
+// For example: https://stackset-myapp-infrastru-pipelinebuiltartifactbuc-1nk5t9zkymh8r.s3-us-west-2.amazonaws.com/scripts/dns-cert-validator/dd2278811c3
```

Comment:
Yeah I looked through them but I think it should be ok since we split by `.` and just pick up the first candidate.

> is that what comes out of our pipeline bucket

Yes. I grabbed it from the pipeline bucket.